src/main/java/org/assertj/core/error/ShouldContainExactly.java

```diff
@@ -92,36 +105,49 @@ public static ErrorMessageFactory shouldContainExactly(Object actual, Object exp
 
   public static ErrorMessageFactory shouldHaveSameSize(Object actual, Object expected, int actualSize,
                                                        int expectedSize, ComparisonStrategy comparisonStrategy) {
-    return StandardComparisonStrategy.instance().equals(comparisonStrategy) ?
-        new ShouldContainExactly(actual, expected, actualSize, expectedSize) :
-        new ShouldContainExactly(actual, expected, actualSize, expectedSize, comparisonStrategy);
+
+    Object[] actualArray = arrayOrIterableToArray(actual);
+    Object[] expectedArray = arrayOrIterableToArray(expected);
+    try {
+      List<Delta<String>> diffs = DIFF.diff(actualArray, expectedArray);
+      final String diffString = diffsAsString(diffs);
+      return StandardComparisonStrategy.instance().equals(comparisonStrategy) ?
+        new ShouldContainExactly(diffString, actualSize, expectedSize) :
+        new ShouldContainExactly(diffString, actualSize, expectedSize, comparisonStrategy);
+    } catch (IOException e) {
+      throw new RuntimeIOException(format("Unable to verify text contents of path:<%s>", actual), e);
+    }
   }
 
-  private ShouldContainExactly(Object actual, Object expected, int actualSize, int expectedSize,
+  private static Object[] arrayOrIterableToArray(Object input) {
```

Comment:
This method belongs naturally to `org.assertj.core.util.Arrays`, let's move it there.
This should have some unit tests too.